crates/sp-consensus-subspace/src/lib.rs

```diff
@@ -335,6 +339,81 @@ impl ChainConstants {
     }
 }
 
+/// Wrapped solution for the purposes of runtime interface.
+#[derive(Debug, Encode, Decode)]
+pub struct WrappedSolution(Solution<FarmerPublicKey, ()>);
+
+impl<RewardAddress> From<&Solution<FarmerPublicKey, RewardAddress>> for WrappedSolution {
+    fn from(solution: &Solution<FarmerPublicKey, RewardAddress>) -> Self {
+        Self(Solution {
+            public_key: solution.public_key.clone(),
+            reward_address: (),
```

Comment:
The root cause is that runtime interface can't be generic.

A bit hacky solution here, but basically the point is that our solution verification doesn't care about reward address at all. Since this is essentially an FFI and Pallet has reward address as generic address, it was tricky to get it to work nicely with above limitation of runtime interface. We know that in normal use addresses are essentially `[u8; 32]` (for now anyway), but in tests they are `u64` and things might change in the future.

I decided to simply ignore it in this context and turn into `()` so we don't have to worry about it anymore. I'm open to better suggestions.